hub-backend/src/main/java/org/panda_lang/hub/user/UserOAuth2DetailsService.kt

```diff
@@ -53,41 +51,43 @@ internal class UserOAuth2DetailsService(
 
         var user: User? = userRepository.findByName(userDetails.name)
 
-        if (user != null ) {
+        user = if (user != null) {
             if (providerName != user.provider) {
                 throw RuntimeException("Looks like you're already signed up with " + user.provider + " account.")
             }
 
-            user = updateExistingUser(user, userDetails)
+            updateExistingUser(user, userDetails)
         } else {
-            user = createNewUser(userRequest, userDetails)
+            createNewUser(userRequest, userDetails)
         }
 
         return UserDetails.of(user, oAuth2User.attributes)
     }
 
     private fun createNewUser(userRequest: OAuth2UserRequest, userDetails: OAuth2UserDetails): User {
         val user = User(
-                userDetails.name,
-                userDetails.displayName ?: StringUtils.EMPTY,
-                StringUtils.EMPTY,
-                StringUtils.EMPTY,
-                userDetails.email,
-                userRequest.clientRegistration.registrationId,
-                userDetails.providerId,
-                userDetails.avatar ?: StringUtils.EMPTY,
-                StringUtils.EMPTY,
-                HashSet()
+            userDetails.name,
+            userDetails.displayName ?: StringUtils.EMPTY,
+            StringUtils.EMPTY,
+            StringUtils.EMPTY,
+            userDetails.email,
+            userRequest.clientRegistration.registrationId,
+            userDetails.providerId,
+            userDetails.avatar ?: StringUtils.EMPTY,
+            StringUtils.EMPTY,
+            HashSet()
         )
 
         return userRepository.save(user)
     }
 
     private fun updateExistingUser(user: User, userDetails: OAuth2UserDetails): User {
-        user.name = userDetails.name
-        user.displayName = userDetails.displayName ?: StringUtils.EMPTY
-        user.avatar = userDetails.avatar ?: StringUtils.EMPTY
-        return userRepository.save(user)
+        return userRepository.save(
+            user.apply {
+                name = userDetails.name
+                displayName = userDetails.displayName ?: StringUtils.EMPTY
+                avatar = userDetails.avatar ?: StringUtils.EMPTY
+            }
```

Comment:
This is similar to aforementioned .also for newly created objects, in this case we use .apply